huggingface_pipelines/text.py

```diff
@@ -0,0 +1,156 @@
+
+
+import json
+import logging
+from datasets import load_dataset
+from sonar.inference_pipelines.text import TextToEmbeddingModelPipeline, EmbeddingToTextModelPipeline
+from evaluate import load
+from dataclasses import dataclass, field
+from typing import List, Dict, Any
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class SonarHFTextToTextPipeline:
+    """
+    A pipeline for encoding text dataset from HuggingFace into embeddings and decoding embeddings back to texts,
+    evaluating the quality using BLEU scores.
+    """
+    encoder_model: str
+    decoder_model: str
+    dataset_name: str
+    dataset_split: str
+    source_lang: str
+    target_lang: str
+    batch_size: int
+    results: List[Dict[str, Any]] = field(default_factory=list)
+    low_bleu_threshold: float = 0.5
```

Comment:
since config remains the main entry point, it would be nice to add some docs here (and motivations as well - just by copying some parts from PR description).